### PR TITLE
Enhancement/3938 idea tracking

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3938

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by felixarntz — CHANGES_REQUESTED** [review]
@eugene-manuilov Mostly looks solid, just a few minor points of feedback.

File: includes/Modules/Idea_Hub.php [hunk 1]
```diff
@@ -864,6 +882,12 @@ private function on_idea_hub_post_status_transition( $new_status, $old_status, $
 		if ( $new_status !== $old_status ) {
 			$this->transients->set( self::IDEA_HUB_LAST_CHANGED, time() );
 		}
+
+		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
+			$this->track_idea_activity( $post->ID, self::ACTIVITY_POST_PUBLISHED );
+		} elseif ( 'publish' !== $new_status && 'publish' === $old_status ) {
+			$this->track_idea_activity( $post->ID, self::ACTIVITY_POST_UNPUBLISHED );
+		}
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/Idea_Hub.php]: Updated.

File: includes/Modules/Idea_Hub.php [hunk 2]
```diff
@@ -1006,4 +1030,48 @@ function( $idea ) use ( $ideas_with_posts ) {
...
+	 * @since n.e.x.t
+	 *
+	 * @param int    $post_id Post ID.
+	 * @param string $type    Activity type.
+	 */
+	private function track_idea_activity( $post_id, $type ) {
+		$service = $this->get_service( 'ideahub' );
+		if ( ! property_exists( $service, 'platforms_properties_ideaActivities' ) ) {
+			return;
+		}
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/Idea_Hub.php]: Ok, updated.

File: includes/Modules/Idea_Hub.php [hunk 3]
```diff
@@ -1006,4 +1030,48 @@ function( $idea ) use ( $ideas_with_posts ) {
...
+		$parent   = $this->get_parent_slug();
+		$activity = new Google_Service_Ideahub_GoogleSearchIdeahubV1alphaIdeaActivity();
+
+		$activity->setIdeas( array( $name ) );
+		$activity->setTopics( array() );
+		$activity->setType( $type );
+
+		if ( 'publish' === $post->post_status ) {
+			$uri = get_permalink( $post );
+			if ( ! empty( $uri ) && is_string( $uri ) ) {
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/Idea_Hub.php]: Yep, doesn't seem to be needed. Removed.

File: includes/Modules/Idea_Hub.php [hunk 4]
```diff
@@ -1006,4 +1030,48 @@ function( $idea ) use ( $ideas_with_posts ) {
...
+				$uri = wp_parse_url( $uri, PHP_URL_PATH );
+				$uri = untrailingslashit( $this->context->get_reference_site_url() ) . $uri;
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply includes/Modules/Idea_Hub.php]: Without using the reference site URL here, I get the following error:

```json
{
  "error": {
    "code": 400,
    "message": "Uri is not subpath of creator uri",
    "errors": [
      {
        "message": "Uri is not subpath of creator uri",
        "domain": "global",
        "reason": "badRequest"
      }
    ],
    "status": "INVALID_ARGUMENT"
  }
}
```

That's why I added this code to make sure the reference site URL is included in the post URL.

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**